emukit/core/parameter_space.py

```diff
@@ -51,3 +56,9 @@ def check_points_in_domain(self, x: np.ndarray) -> np.ndarray:
             # Set in_domain to be False if this parameter or any previous parameter is out of domain
             in_domain = np.all([in_domain, param_in_domain], axis=0)
         return in_domain
+
+    def append(self, new_parameter):
```

Comment:
This is a bit scary - changing space on the fly. I'd like to think of parameter space as an immutable thing once defined.

Is it used? Can we avoid using it?

Reply:
Removed